crypto/fipsmodule/dh/check.c

```diff
@@ -203,24 +202,6 @@ int DH_check(const DH *dh, int *out_flags) {
     if (!BN_is_one(t2)) {
       *out_flags |= DH_CHECK_INVALID_Q_VALUE;
     }
-  } else if (BN_is_word(dh->g, DH_GENERATOR_2)) {
-    l = BN_mod_word(dh->p, 24);
```

Comment:
You should update the documentation at the top of this function as I think it's out of date now.  I would also add a small note (that you put in the PR) in the header file to explain why the checks are they way they are.